Modules/Utils/Sources/PocketCastsUtils/General/KPMSearch.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+// An implementation of the Knuth-Morris-Pratt algorithm
+public class KMPSearch {
+    private var pattern: [Character]
+    private var lps: [Int]
+
+    public init(pattern: String) {
+        self.pattern = Array(pattern.lowercased())
+        self.lps = Array(repeating: 0, count: pattern.count)
+        computeLPSArray()
+    }
+
+    private func computeLPSArray() {
+        guard lps.count > 0 else {
```

Comment:
I would use `if lps.isEmpty` directly.
In general with collections, `isEmpty` is more efficient than `count` as it doesn't loop trough the whole list.

Reply:
Done in 4fee0685bdb7e08cb29169c2c6e15d3b42b0732f